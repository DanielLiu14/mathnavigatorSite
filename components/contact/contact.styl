@import "../static.styl"

#view-contact-container
  width: containerWidthPct
  max-width: containerWidthPx1
  margin: 0 auto spacing24x

  & .section
    margin: spacing4x 0 spacing8x

  & h1
    margin: spacing8x 0 spacing6x
    line-height: lineHeight2
    font-family: fontFamilyRockwell
    font-size: fontSize2
    color: colorPrimary

  & h2
    margin: spacing6x 0 spacing2x
    line-height: lineHeight1
    font-size: fontSize1
    color: colorPrimary

  & h3
    line-height: lineHeight0
    font-size: fontSize0
    font-weight: 400
    color: colorBlackDark

#view-contact-container

  & .contact-input-container
    margin: spacing2x 0

#view-contact-container
  & .section.interested h2
    margin-bottom: spacing2x

  & .section.additional
    margin-bottom: spacing2x

    & h2
      margin-bottom: spacing2x

    & textarea
      width: propUnit16x
      height: propUnit4x
      padding: spacing2x
      box-sizing: border-box
      font-family: fontFamilyRoboto
      font-size: fontSize0
      line-height: lineHeight0
      resize: none
      outline: colorPrimary

      &:focus
        outline-color: colorTurquoiseLight

  & .section.submit
    position: relative
    width: propUnit16x

    & p
      margin: 0
      line-height: lineHeight0
      font-size: fontSize0
      color: colorBlackDark

    & .list-fields-remaining
      margin-top: spacing2x
      font-size: fontSize0
      display: none

      &.show
        display: block

      & ul
        margin: 0

    & button
      position: absolute
      right: 0
      top: 0
      background-color: colorGrayLight
      color: colorBlackDark
      cursor: initial

      &:hover
        filter: none

      &.active
        background-color: colorOrange
        color: colorWhite
        cursor: pointer

        &:hover
          filter: brightness(110%)

  & .section.errors li
    line-height: lineHeight0
    font-size: fontSize0
    color: colorOrange

    &:first-child
      color: colorBlack
      margin-bottom: spacing1x
