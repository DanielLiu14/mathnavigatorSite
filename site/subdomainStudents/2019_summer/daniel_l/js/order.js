console.log("My Javascript works");

function init() {
  console.log("jQuery ready!");
  var flavor = window.localStorage.getItem("flavor");
  $("select.milk-tea").val(flavor);
  var ice = window.localStorage.getItem("ice");
  $("select.ice-amount").val(ice);
  var sugar = window.localStorage.getItem("sugar");
  $("select.sugar-amount").val(sugar);  
  var bubbles = window.localStorage.getItem("bubbles");
  var isChecked = bubbles =="true";
  $("input.bubbles").prop("checked", isChecked);
  var grassJelly = window.localStorage.getItem("grass-jelly");
  var isChecked2 = grassJelly =="true";
  $("input.grass-jelly").prop("checked", isChecked);
  var pudding = window.localStorage.getItem("pudding");
  var isChecked3 = pudding =="true";
  $("input.pudding").prop("checked", isChecked);

  // Setup on-click handler
  $("button").on("click", handleClick);
  $(".milk-tea").on("change", handleSelectChange);
  $(".ice-amount").on("change", handleSelectChange);
  $(".sugar-amount").on("change", handleSelectChange);
  $(".bubbles").on("change", handleCheckboxChange);
  $(".grass-jelly").on("change", handleCheckboxChange);
  $(".pudding").on("change", handleCheckboxChange);
}


function handleSelectChange(e) {
  var targetVal = e.target.value;
  var targetClass = e.target.className;
  if (targetClass == "milk-tea"){
	window.localStorage.setItem("flavor", targetVal);
  }
  if (targetClass == "ice-amount"){
	window.localStorage.setItem("ice", targetVal);
  }
  if (targetClass == "sugar-amount"){
	window.localStorage.setItem("sugar", targetVal);
  }
}

function handleCheckboxChange(e) {
  var targetVal = e.target.checked;
  var targetClass = e.target.className;
  if (targetClass == "bubbles"){
    window.localStorage.setItem("bubbles", targetVal);
  }
  if (targetClass == "grass-jelly"){
    window.localStorage.setItem("grass-jelly", targetVal);
  }
  if (targetClass == "pudding"){
    window.localStorage.setItem("pudding", targetVal);
  }
}

function handleClick() {
  var milkTeaSelected = $(".milk-tea").val();
  var iceAmount = $(".ice-amount").val();
  var sugarAmount = $(".sugar-amount").val();
  var name = $(".enterName").val();
  var url = "html/receipt.html?flavor="+milkTeaSelected+"&ice="+iceAmount+"&sugar="+sugarAmount+"&name="+name;
  if ($(".bubbles").prop("checked") == true) {
	  url += "&toppings=bubbles";
  }
  if ($(".grass-jelly").prop("checked") == true) {
	  url += "&toppings=grassjelly";
  }
  if ($(".pudding").prop("checked") == true) {
	  url += "&toppings=pudding";
  }
  console.log(url);
  window.location = url;
}


$(document).ready(init);