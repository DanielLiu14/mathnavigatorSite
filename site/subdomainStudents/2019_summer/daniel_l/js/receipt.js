console.log("My Javascript works");

var toppings = [];

var imgMap = {
  brownsugar: "../assets/caramel.jpg",
  jasmine: "../assets/cappuccino.jpg",
  oolong: "../assets/black_coffee.jpg",
  thai: "../assets/latte.jpg",
  taro: "../assets/cold_brew.jpg"
};

var flavorStringMap = {
  brownsugar: "Caramel Coffee",
  jasmine: "Cappuccino",
  oolong: "Black Coffee",
  thai: "Latte",
  taro: "Cold Brew"
};

var iceStringMap = {
  full: "More ice",
  half: "Normal ice",
  none: "No Ice"
};

var sugarStringMap = {
  hundred: "Two packets of sugar",
  fifty: "One packet of sugar",
  twentyfive: "No sugar"
};

var toppingStringMap = {
  bubbles: "Foam",
  grassjelly: "Expresso",
  pudding: "Cream"
}

var priceMap = {
  brownsugar: 4,
  jasmine: 2.5,
  oolong: 2.5,
  thai: 3,
  taro: 3
};

function init() {
  console.log("jQuery ready!");
  var map = convertSearch(window.location.search);
  var name1 = map["name"];
  var name2 = name1.replace("%20"," ");
  var totalPrice = priceMap[map["flavor"]]+.5*toppings.length;
  console.log(JSON.stringify(map));
  console.log(JSON.stringify(toppings));
  $("img").attr("src",imgMap[map["flavor"]]);
  $(".greetings").text("Hi " + name2 + ", thanks for ordering at Starbucks! This is your receipt:");
  $(".flavor").text(flavorStringMap[map["flavor"]]);
  $(".ice").text("Ice Amount: " + iceStringMap[map["ice"]]);
  $(".sugar").text("Sugar Amount: " + sugarStringMap[map["sugar"]]);
  $(".price").text("Price: " + totalPrice + " dollars");
  if (toppings.length == 0){
    $(".toppings").text("Toppings: No Toppings");
  }
  if (toppings.length == 1){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]);
  }
  if (toppings.length == 2){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]+", "+toppingStringMap[toppings[1]]);
  }
  if (toppings.length == 3){
	$(".toppings").text("Toppings: "+toppingStringMap[toppings[0]]+", "+toppingStringMap[toppings[1]]+", "+toppingStringMap[toppings[2]]);
  }
}

$(document).ready(init);

function convertSearch(search) {
  var map = {};
  search = search || "";
  search = search.substring(1);
  var pairList = search.split("&");
  for (var i=0; i < pairList.length; i++) {
    var pair = pairList[i];
    var split = pair.split("=");
    var key = split[0];
    var value = split[1];
	if(key == "toppings") {
		toppings.push(value);
	} else {
		map[key] = value;
	}
  }
  return map;
}