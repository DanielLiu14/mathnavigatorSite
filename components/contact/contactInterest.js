'use strict';
require('./contactInterest.styl');
import { pull } from 'lodash';
import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { Modal } from '../modals/modal.js';
import {
  createFullClassObj,
  getProgramClass,
  getClassesBySemester,
  getSemester,
	getSemesterIds
} from '../repos/mainRepo.js';
const classnames = require('classnames');

export class ContactInterestSection extends React.Component {
  constructor(props) {
    super(props);
    var interested = [];

    var parsed = parseQuery();
    if (parsed.interest) {
      interested.push(parsed.interest);
    }

    this.state = {
      showModal: false,
      interested: interested
    };

    this.onSelectProgram = this.onSelectProgram.bind(this);
    this.onToggleShowModal = this.onToggleShowModal.bind(this);
  }

  onToggleShowModal() {
    const show = this.state.showModal;
    var newShow = !show;
    this.setState({
      showModal: newShow
    });
  }

  onSelectProgram(classKey, selected) {
    var interestedList = this.state.interested;
    if (selected) {
      interestedList.push(classKey);
    } else {
      interestedList = pull(interestedList, classKey);
    }
    this.setState({
      interested: interestedList
    });

    if (this.props.onUpdate) {
      this.props.onUpdate(interestedList);
    }
  }

  render() {
    const interestedClasses = this.state.interested || [];
    const interestedSection = generateInterested(interestedClasses, this.onToggleShowModal);
    const modalContent = <InterestModal
                  onSelectProgram={this.onSelectProgram}
                  interested={interestedClasses}
                  onDismiss={this.onToggleShowModal}/>;
    return (
      <div>
        <h2>Interested Programs</h2>
        {interestedSection}
        <Modal content={modalContent}
          show={this.state.showModal}
          withClose={true}
          onDismiss={this.onToggleShowModal}/>
      </div>
    );
  }
}

class InterestModal extends React.Component {
  render() {
    const interestedClasses = this.props.interested || [];
    var interestClassMap = {};
    interestedClasses.forEach((classKey) => {
      interestClassMap[classKey] = true;
    });
    const numClasses = interestedClasses.length;
    const onSelectProgram = this.props.onSelectProgram;

    const semesterIds = getSemesterIds();
		const sections = semesterIds.map(function(semesterId, index) {
      var semester = getSemester(semesterId);
			var classes = getClassesBySemester(semesterId);
      const list = createInterestItems(classes, interestClassMap, onSelectProgram);
			return (
        <div key={index}>
          <h2>{semester.title}</h2>
          <ul>
            {list}
          </ul>
        </div>
			);
		});

    var selectedLineClassNames = classnames("selected-line", {
      highlight: numClasses > 0
    });
    var selectedLineText = numClasses;
    selectedLineText += (numClasses == 1 ? " class " : " classes ");
    selectedLineText += "selected";

    return (
      <div id="interest-modal">
        <h1>Interested Programs</h1>
        <div id="interest-headers">
          <div className="header class-name">Class</div>
          <div className="header times">Times</div>
          <div className="header start-date">Starting Date</div>
        </div>
        <div id="interest-view" className="use-scrollbar">
          {sections}
        </div>
        <div className={selectedLineClassNames}>{selectedLineText}</div>
        <button className="btn-done" onClick={this.props.onDismiss}>Done</button>
      </div>
    );
  }
}

class InterestItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      highlighted: this.props.selected || false
    }
    this.handleSelected = this.handleSelected.bind(this);
  }

  handleSelected(classKey) {
    var selected = this.state.highlighted;
    var newSelected = !selected;
    this.setState({
      highlighted: newSelected
    });
    if (this.props.onSelect) {
      var classKey = this.props.classObj.key;
      this.props.onSelect(classKey, newSelected);
    }
  }

  render() {
    const classObj = createFullClassObj(this.props.classObj);
    const className = classObj.fullClassName;
    const startingDate = classObj.startDate;

    const liClassNames = classnames("", {
      "highlight": this.state.highlighted
    });

    const times = classObj.times.map((time, index) =>
      <div key={index}>{time}</div>
    );

    return (
      <li className={liClassNames}>
        <input type="checkbox" name="interest" value="program"
              onChange={this.handleSelected}
              checked={this.props.selected}/>
        <div className="class-name">{className}</div>
        <div className="times">{times}</div>
        <div className="start-date">{startingDate}</div>
      </li>
    );
  }
}

/* Helper Functions */
function generateInterested(interestedList, onClick) {
  var nonEmpty = interestedList.length > 0;
  var interestedButtonText = nonEmpty ? "Select More Programs..." : "Select Programs...";
  if (nonEmpty) {
    var numClasses = interestedList.length;
    var numClassText = numClasses == 1 ? numClasses + " class." : numClasses + " classes."

    const selectedText = interestedList.map(function(classKey, index) {
      var pair = getProgramClass(classKey);
      var programTitle = pair.programObj.title;
      var className = pair.classObj.className;
      var fullName = className ? programTitle + " " + className : programTitle;
      var url = "/class/" + classKey;
      return (
        <li key={index}><Link to={url}>{fullName}</Link></li>
      );
    });

    return (
      <div id="contact-section-interested">
        <p>
          You have selected {numClassText}<br/>
          {selectedText}
        </p>
        <button className="inverted" onClick={onClick}>
          {interestedButtonText}
        </button>
      </div>
    );
  } else {
    return (
      <div id="contact-section-interested">
        <p>
          Please select at least one Program.
        </p>
        <button className="inverted" onClick={onClick}>
          {interestedButtonText}
        </button>
      </div>
    );
  }
}

function createInterestItems(listClasses, interestedMap, onSelect) {
  return listClasses.map(function(classObj, index) {
    var isSelected = interestedMap[classObj.key] || false;
    return (
      <InterestItem key={index}
                classObj={classObj}
                selected={isSelected}
                onSelect={onSelect}/>
    );
  });
}

function parseQuery() {
  var hash = window.location.hash;
  var i = hash.indexOf("?");
  var parsed = {};
  if (i > 0) {
    hash = hash.slice(i + 1);

    // parse Query String
    var params = hash.split("&");
    for (var i = 0; i < params.length; i++) {
      var pair = params[i].split("=");
      var pairKey = decodeURIComponent(pair[0]);
      var pairValue = decodeURIComponent(pair[1]);
      parsed[pairKey] = pairValue;
    }
  }
  return parsed;
}
