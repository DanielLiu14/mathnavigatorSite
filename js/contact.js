'use strict';
require('./../styl/contact.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { ContactInputForm } from './contactInputForm.js';
const classnames = require('classnames');


export class ContactPage extends React.Component {
	componentDidMount() {
	  window.scrollTo(0, 0);
	}

	render() {
		return (
      <div id="view-contact">
        <div id="view-contact-container">
          <h1>Contact Us</h1>
					<ContactInputForm/>
        </div>
      </div>
		);
	}
}
