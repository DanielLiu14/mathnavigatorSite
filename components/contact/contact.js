'use strict';
require('./contact.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { ContactForm } from './contactForm.js';
const classnames = require('classnames');

export class ContactPage extends React.Component {
	constructor(props) {
		super(props);

		const { match, location, history } = this.props;
		this.interested = [];
		var parsed = parseQuery((location || {}).search);
		if (parsed && parsed.interest) {
			this.interested.push(parsed.interest);
		}
	}

	componentDidMount() {
		if (process.env.NODE_ENV === 'production') {
			mixpanel.track("contact");
		}
	}

	render() {
		return (
      <div id="view-contact">
        <div id="view-contact-container">
          <h1>Contact Us</h1>
					<h3>
						Please fill the following form<br/>
						to register for a program or to contact us.
					</h3>
					<ContactForm startingInterest={this.interested}/>
        </div>
      </div>
		);
	}
}

function parseQuery(hash) {
	if (!hash) {
		return "";
	}
  var i = hash.indexOf("?");
  var parsed = {};
  if (i >= 0) {
    hash = hash.slice(i + 1);

    // parse Query String
    var params = hash.split("&");
    for (var i = 0; i < params.length; i++) {
      var pair = params[i].split("=");
      var pairKey = decodeURIComponent(pair[0]);
      var pairValue = decodeURIComponent(pair[1]);
      parsed[pairKey] = pairValue;
    }
  }
  return parsed;
}
