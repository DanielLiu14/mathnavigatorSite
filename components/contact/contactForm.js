'use strict';
require('./contact.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { ContactInterestSection } from './contactInterest.js';
import {
  EmailModal,
  STATE_NONE,
  STATE_EMPTY,
  STATE_LOADING,
  STATE_SUCCESS,
  STATE_FAIL
} from '../modals/emailModal.js';
import { sendEmail, sendTestEmail } from '../repos/emailRepo.js';
import {
  EmailCheck,
  SchoolCheck,
  GradeCheck,
  NameCheck,
  PhoneCheck
} from '../forms/formInputChecks.js';
import { FormInput } from '../forms/formInput.js';
import { pull } from 'lodash';
import { Modal } from '../modals/modal.js';
const classnames = require('classnames');

export class ContactForm extends React.Component {
	constructor(props) {
    super(props);

    var interested = [];
    var parsed = parseQuery();
    if (parsed && parsed.interest) {
      interested.push(parsed.interest);
    }

		this.state = {
      submitState: STATE_NONE,
			studentFirstName: "",
			studentLastName: "",
			studentGrade: 0,
			studentSchool: "",
			studentPhone: "",
			studentEmail: "",
			guardFirstName: "",
			guardLastName: "",
			guardPhone: "",
			guardEmail: "",
			interestedPrograms: interested,
			additionalText: "",
      generatedEmail: null
		};

    this.handleSubmit = this.handleSubmit.bind(this);
    this.onSubmitSuccess = this.onSubmitSuccess.bind(this);
    this.onSubmitFail = this.onSubmitFail.bind(this);
    this.dismissModal = this.dismissModal.bind(this);

		this.getInputInfo = this.getInputInfo.bind(this);
		this.updateCb = this.updateCb.bind(this);
    this.updateInterested = this.updateInterested.bind(this);
		this.updateTextArea = this.updateTextArea.bind(this);

    this.checkAllInputs = this.checkAllInputs.bind(this);
  }

	updateCb(propertyName, newValue) {
		var obj = {};
		obj[propertyName] = newValue;
		this.setState(obj);
	}

  updateInterested(classKey, selected) {
    var interestedList = this.state.interestedPrograms;
    if (selected) {
      interestedList.push(classKey);
    } else {
      interestedList = pull(interestedList, classKey);
    }
    this.setState({
      interestedPrograms: interestedList
    });
  }

	updateTextArea(event) {
		this.updateCb("additionalText", event.target.value);
	}

	render() {
    const submitState = this.state.submitState;
    const modalContent = <EmailModal
                            loadingState={submitState}
                            failText={this.state.generatedEmail}/>;
    const showModal = submitState != STATE_NONE;

    const formCompleted = this.checkAllInputs();
    const submitBtnClass = classnames({active: formCompleted});
    const onHandleSubmit = formCompleted ? this.handleSubmit : undefined;

		return (
      <div>
        <Modal content={modalContent}
                show={showModal}
                persistent={true}
                onDismiss={this.dismissModal}/>
        <div className="section input">
          <h2>Student Information</h2>
          <div className="contact-input-container">
            <FormInput addClasses="student-fname" title="First Name" propertyName="studentFirstName"
                  onUpdate={this.updateCb} validator={NameCheck}/>
            <FormInput addClasses="student-lname" title="Last Name" propertyName="studentLastName"
                  onUpdate={this.updateCb} validator={NameCheck}/>
          </div>
          <div className="contact-input-container">
            <FormInput addClasses="student-grade" title="Grade" propertyName="studentGrade"
                  onUpdate={this.updateCb} validator={GradeCheck}/>
            <FormInput addClasses="student-school" title="School" propertyName="studentSchool"
                  onUpdate={this.updateCb} validator={SchoolCheck}/>
          </div>
          <div className="contact-input-container">
            <FormInput addClasses="student-phone" title="Phone" propertyName="studentPhone"
                  onUpdate={this.updateCb} validator={PhoneCheck}/>
            <FormInput addClasses="student-email" title="Email" propertyName="studentEmail"
                  onUpdate={this.updateCb} validator={EmailCheck}/>
          </div>

          <h2>Guardian Information</h2>
          <div className="contact-input-container">
            <FormInput addClasses="guard-fname" title="First Name" propertyName="guardFirstName"
                  onUpdate={this.updateCb} validator={NameCheck}/>
            <FormInput addClasses="guard-lname" title="Last Name" propertyName="guardLastName"
                  onUpdate={this.updateCb} validator={NameCheck}/>
          </div>
          <div className="contact-input-container">
            <FormInput addClasses="guard-phone" title="Phone" propertyName="guardPhone"
                  onUpdate={this.updateCb} validator={PhoneCheck}/>
            <FormInput addClasses="guard-email" title="Email" propertyName="guardEmail"
                  onUpdate={this.updateCb} validator={EmailCheck}/>
          </div>
        </div>
				<div className="section interested">
					<ContactInterestSection interested={this.state.interestedPrograms}
                                  onUpdate={this.updateInterested}/>
				</div>

				<div className="section additional">
					<h2>Additional Information</h2>
					<textarea onChange={this.updateTextArea} placeholder="(Optional)"/>
        </div>

        <div className="section submit">
          <div className="submit-container">
            <p>
              Information will be sent to:<br/>
              <a>andymathnavigator@gmail.com</a>
            </p>
            <button className={submitBtnClass} onClick={onHandleSubmit}>
              Submit
            </button>
          </div>
        </div>
      </div>
		);
	}

	getInputInfo() {
		return {
			studentFirstName: this.state.studentFirstName,
			studentLastName: this.state.studentLastName,
			studentGrade: this.state.studentGrade,
			studentSchool: this.state.studentSchool,
			studentPhone: this.state.studentPhone,
			studentEmail: this.state.studentEmail,
			guardFirstName: this.state.guardFirstName,
			guardLastName: this.state.guardLastName,
			guardPhone: this.state.guardPhone,
			guardEmail: this.state.guardEmail,
			interestedPrograms: this.state.interestedPrograms,
			additionalText: this.state.additionalText
		};
	}

  checkAllInputs() {
    return NameCheck.validate(this.state.studentFirstName)
                    // && NameCheck.validate(this.state.studentLastName)
                    // && GradeCheck.validate(this.state.studentGrade)
                    // && SchoolCheck.validate(this.state.studentSchool)
                    // && PhoneCheck.validate(this.state.studentPhone)
                    // && EmailCheck.validate(this.state.studentEmail)
                    // && NameCheck.validate(this.state.guardFirstName)
                    // && NameCheck.validate(this.state.guardLastName)
                    // && PhoneCheck.validate(this.state.guardPhone)
                    // && EmailCheck.validate(this.state.guardEmail)
                    && this.state.interestedPrograms.length > 0;
  }

	handleSubmit(event) {
    event.preventDefault();

		var inputInfo = this.getInputInfo();
		const emailMessage = generateEmailMessage(inputInfo);

    console.log("Sending email... " + emailMessage);
    this.setState({
      submitState: STATE_LOADING,
      generatedEmail: emailMessage
    });

    sendEmail(emailMessage, this.onSubmitSuccess, this.onSubmitFail);
	}

	onSubmitSuccess() {
    setTimeout(() => {
      this.setState({ submitState: STATE_EMPTY });
      setTimeout(() => {
        console.log("Email success!");
        this.setState({ submitState: STATE_SUCCESS });
      }, 400);
    }, 3600);

	}

	onSubmitFail() {
    setTimeout(() => {
      this.setState({ submitState: STATE_EMPTY });
      setTimeout(() => {
        console.log("Email fail!");
        this.setState({ submitState: STATE_FAIL });
      }, 400);
    }, 3600);
	}

  dismissModal() {
    console.log("Dismiss modal");
    this.setState({
      submitState: STATE_LOADING
    });
  }
}

/* Helper functions */

function parseQuery() {
  var hash = window.location.hash;
  var i = hash.indexOf("?");
  var parsed = {};
  if (i > 0) {
    hash = hash.slice(i + 1);

    // parse Query String
    var params = hash.split("&");
    for (var i = 0; i < params.length; i++) {
      var pair = params[i].split("=");
      var pairKey = decodeURIComponent(pair[0]);
      var pairValue = decodeURIComponent(pair[1]);
      parsed[pairKey] = pairValue;
    }
  }
  return parsed;
}

function generateEmailMessage(info) {
	if (!info) {
		return null;
	}
	return [
    "<html>",
    "<body>",
    "<h1>To Math Navigator,</h1>",
    "<h2>Contact Us Page</h2>",
    "<h3>Student: " + info.studentFirstName + "	&nbsp; " + info.studentLastName + "</h3>",
    "<h3>Grade:" + info.studentGrade + "</h3>",
    "<h3>School: " + info.studentSchool + "</h3>",
    "<h3>Phone: " + info.studentPhone + "</h3>",
    "<h3>Email: " + info.studentEmail + "</h3>",
    "<br/>",
    "<h3>Guardian: " + info.guardFirstName + "	&nbsp; " + info.guardLastName + "</h3>",
    "<h3>Phone: " + info.guardPhone + "</h3>",
    "<h3>Email: " + info.guardEmail + "</h3>",
    "<br/>",
    "<p>Interested Programs: " + JSON.stringify(info.interestedPrograms) + "</p>",
    "<p>Additional Info: " + info.additionalText + "</p>",
    "</body>",
    "</html>"
	].join("\n");
}
