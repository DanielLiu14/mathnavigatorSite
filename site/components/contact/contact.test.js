import React from "react";
import Enzyme, { shallow } from "enzyme";
import { HashRouter as Router } from 'react-router-dom';
import { ContactPage } from "./contact.js";
import { ContactForm } from "./contactForm.js";
import { ContactInterestSection } from "./contactInterest.js";
import renderer from 'react-test-renderer';

describe("test", () => {

  test("renders", () => {
    const component = shallow(<ContactPage/>);
    expect(component.exists()).toBe(true);
    expect(component.find("h1").text()).toBe("Contact Us");
    expect(component.exists("ContactForm")).toBe(true);
  });

  test("parse location query", () => {
    var location = {
      search: "contact?interest=asdf"
    };
    var component = shallow(<ContactPage location={location}/>);
    var form = component.find("ContactForm");
    expect(form.prop('startingInterest')).toEqual(["asdf"]);
  });

  test("renders form", () => {
    const formComponent = shallow(<ContactForm/>);
    expect(formComponent.exists()).toBe(true);
    expect(formComponent.exists(".submit-container button")).toBe(true);

    const interestComponent = formComponent.find("ContactInterestSection");
    expect(interestComponent.exists()).toBe(true);
  });

  test("renders form with parameters", () => {
    var interested = ["asdf"];
    const formComponent = shallow(<ContactForm startingInterest={interested}/>);
    expect(formComponent.state('interestedPrograms')).toEqual(interested);

    const interestComponent = formComponent.find("ContactInterestSection");
    expect(interestComponent.prop('interested')).toEqual(interested);
  });

  test("renders interest", () => {
    const component = shallow(<ContactInterestSection/>);
    expect(component.exists()).toBe(true);
    expect(component.find("h2").text()).toBe("Interested Programs");
  });

  // test for updating interest section -> updates form.interestedPrograms

  test("snapshot", () => {
    const tree = renderer.create(
      <Router>
        <ContactPage/>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
