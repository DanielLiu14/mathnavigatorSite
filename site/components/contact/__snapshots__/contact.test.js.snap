// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  id="view-contact"
>
  <div
    id="view-contact-container"
  >
    <h1>
      Contact Us
    </h1>
    <h3>
      Please fill the following form
      <br />
      to register for a program or to contact us.
    </h3>
    <div>
      <div
        className="modal-view"
      >
        <div
          className="modal-overlay"
        />
        <div
          className="modal"
        >
          <div
            className=""
            id="email-modal"
          >
            <div
              className="container loading"
            >
              <h1>
                Sending...
              </h1>
              <div
                className="content"
              >
                <div
                  className="dots-container"
                >
                  <div
                    className="dots"
                  >
                    <div
                      className="dot dot1"
                    />
                    <div
                      className="dot dot2"
                    />
                    <div
                      className="dot dot3"
                    />
                  </div>
                  <img
                    src="test-file-stub"
                  />
                </div>
              </div>
            </div>
            <div
              className="container success"
            >
              <h1>
                Success
              </h1>
              <div
                className="content"
              >
                <div
                  className="img-wrapper"
                >
                  <img
                    src="test-file-stub"
                  />
                </div>
                <p>
                  Thank you for contacting us.
                  <br />
                  You will hear from us soon!
                  <br />
                  <a
                    href="#/"
                    onClick={[Function]}
                  >
                    Back to Home
                  </a>
                </p>
              </div>
            </div>
            <div
              className="container fail"
            >
              <h1>
                Email request failed
              </h1>
              <div
                className="content"
              >
                <img
                  src="test-file-stub"
                />
                <p>
                  Unfortunately, our email servers are not working at the moment.
                  <br />
                  Please try again later or
                  <br />
                  send your message to 
                  <b>
                    andymathnavigator@gmail.com
                  </b>
                  .
                  <br />
                  We apologize for any inconveniences.
                </p>
                <a
                  href="#/"
                  onClick={[Function]}
                >
                  Back to Home
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="section input"
      >
        <h2>
          Student Information
        </h2>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input student-fname"
          >
            <label>
              First Name
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
          <div
            className="form-input student-lname"
          >
            <label>
              Last Name
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input student-grade"
          >
            <label>
              Grade
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
          <div
            className="form-input student-school"
          >
            <label>
              School
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input student-phone"
          >
            <label>
              Phone
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
          <div
            className="form-input student-email"
          >
            <label>
              Email
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
        <h2>
          Guardian Information
        </h2>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input guard-fname"
          >
            <label>
              First Name
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
          <div
            className="form-input guard-lname"
          >
            <label>
              Last Name
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input guard-phone"
          >
            <label>
              Phone
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
          <div
            className="form-input guard-email"
          >
            <label>
              Email
            </label>
            <input
              onChange={[Function]}
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
        <div
          className="contact-input-container"
        >
          <div
            className="form-input guard-social-wechat"
          >
            <label>
              WeChat username
            </label>
            <input
              onChange={[Function]}
              placeholder="(Optional)"
              value=""
            />
            <label>
              
            </label>
          </div>
        </div>
      </div>
      <div
        className="section interested"
      >
        <div>
          <h2>
            Interested Programs
          </h2>
          <div
            id="contact-section-interested"
          >
            <p>
              Please select at least one Program.
            </p>
            <button
              className="inverted"
              onClick={[Function]}
            >
              Select Programs...
            </button>
          </div>
          <div
            className="modal-view"
          >
            <div
              className="modal-overlay"
              onClick={[Function]}
            />
            <div
              className="modal"
            >
              <button
                className="close-x"
                onClick={[Function]}
              >
                <img
                  src="test-file-stub"
                />
              </button>
              <div
                id="interest-modal"
              >
                <h1>
                  Interested Programs
                </h1>
                <div
                  id="interest-headers"
                >
                  <div
                    className="header class-name"
                  >
                    Class
                  </div>
                  <div
                    className="header times"
                  >
                    Times
                  </div>
                  <div
                    className="header start-date"
                  >
                    Starting Date
                  </div>
                </div>
                <div
                  className="use-scrollbar"
                  id="interest-view"
                />
                <div
                  className="selected-line"
                >
                  0 classes selected
                </div>
                <button
                  className="btn-done"
                  onClick={[Function]}
                >
                  Done
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="section additional"
      >
        <h2>
          Additional Information
        </h2>
        <textarea
          onChange={[Function]}
          placeholder="(Optional)"
        />
      </div>
      <div
        className="section submit"
      >
        <div
          className="submit-container"
        >
          <button
            className=""
          >
            Submit
          </button>
          <p>
            Information will be sent to:
            <br />
            <a>
              andymathnavigator@gmail.com
            </a>
          </p>
        </div>
      </div>
      <div
        className="section errors"
      >
        <ul>
          <li>
            Please correctly fill the form in order to submit!
          </li>
          <li>
            Please completely fill your student information.
          </li>
          <li>
            Please completely fill your guardian information.
          </li>
          <li>
            Pick at least one program.
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
